SYCL/ESIMD/api/functional/ctors/ctor_fill_core.cpp

```diff
@@ -0,0 +1,153 @@
+  // it's a step value.
+  passed &=
+      ctors::run_verification<ctors::initializer, ctors::init_val::min_half,
+                              ctors::init_val::zero>(queue, two_dims,
+                                                     char_int_types);
+
+  passed &= ctors::run_verification<ctors::initializer, ctors::init_val::zero,
+                                    ctors::init_val::positive>(queue, two_dims,
+                                                               char_int_types);
+  passed &=
+      ctors::run_verification<ctors::initializer, ctors::init_val::min_half,
+                              ctors::init_val::positive>(queue, two_dims,
+                                                         char_int_types);
+
+  passed &= ctors::run_verification<ctors::var_dec, ctors::init_val::min_half,
+                                    ctors::init_val::zero>(queue, two_dims,
+                                                           char_int_types);
+  passed &= ctors::run_verification<ctors::var_dec, ctors::init_val::zero,
+                                    ctors::init_val::positive>(queue, two_dims,
+                                                               char_int_types);
+  passed &= ctors::run_verification<ctors::var_dec, ctors::init_val::min_half,
+                                    ctors::init_val::positive>(queue, two_dims,
+                                                               char_int_types);
+
+  passed &=
+      ctors::run_verification<ctors::rval_in_express, ctors::init_val::min_half,
+                              ctors::init_val::zero>(queue, two_dims,
+                                                     char_int_types);
+  passed &=
+      ctors::run_verification<ctors::rval_in_express, ctors::init_val::zero,
+                              ctors::init_val::positive>(queue, two_dims,
+                                                         char_int_types);
+  passed &=
+      ctors::run_verification<ctors::rval_in_express, ctors::init_val::min_half,
+                              ctors::init_val::positive>(queue, two_dims,
+                                                         char_int_types);
+
+  passed &= ctors::run_verification<ctors::const_ref, ctors::init_val::min_half,
+                                    ctors::init_val::zero>(queue, two_dims,
+                                                           char_int_types);
+  passed &= ctors::run_verification<ctors::const_ref, ctors::init_val::zero,
+                                    ctors::init_val::positive>(queue, two_dims,
+                                                               char_int_types);
+  passed &= ctors::run_verification<ctors::const_ref, ctors::init_val::min_half,
+                                    ctors::init_val::positive>(queue, two_dims,
+                                                               char_int_types);
+#ifdef SIMD_RUN_TEST_WITH_VECTOR_LEN_32
+  const auto all_dims = values_pack<1, 8, 16, 32>();
```

Comment:
Sorry, if I duplicate this comment. The first one disappeared, I don't see it.
Wouldn't it be better way to say something like this:
```suggestion
#ifdef SIMD_RUN_TEST_WITH_VECTOR_LEN_32
  const auto all_dims = values_pack<1, 8, 16, 32>();
#else
  const auto all_dims = values_pack<1, 8, 16>();
#endif  
```

Reply:
I've applied this comment in this commit: a3b04bede68dfb78f3144647d94e7847fa385525